### PR TITLE
Virsh desc add cases

### PR DESCRIPTION
**Pull request by liuzzfnst**
Set the tittle and description with different string in virsh desc.
    
Currently the string for testing set tittle and description is the
same in virsh desc, maybe make then different is better.
Add the new case for '--edit' option.

### CODE REVIEW

**Comment by liuzzfnst** [comment]
The test result on libvirt-1.2.13 fc20.x86_64
\# ./run -t libvirt --no-downloads -k --tests virsh.desc
SETUP: PASS (0.20 s)
DATA DIR: /var/lib/virt_test
DEBUG LOG: /root/autotest/client/tests/virt/logs/run-2015-07-07-11.09.18/debug.log
TESTS: 22
(1/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.no_option: PASS (44.20 s)
(2/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.domain_ID_no_option: PASS (24.46 s)
(3/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.domain_UUID_no_option: PASS (36.36 s)
(4/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.title_option: PASS (46.37 s)
(5/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.live_desc: PASS (32.47 s)
(6/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.current_desc: PASS (32.29 s)
(7/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.config_desc: PASS (66.96 s)
(8/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.live_config_desc: PASS (47.50 s)
(9/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.live_title: PASS (33.57 s)
(10/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.current_title: PASS (43.52 s)
(11/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.config_title: PASS (63.90 s)
(12/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.live_config_title: PASS (32.89 s)
(13/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.edit_option: PASS (33.45 s)
(14/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.new_desc_option: PASS (42.78 s)
(15/22) type_specific.io-github-autotest-libvirt.virsh.desc.positive_test.transient_live: PASS (30.81 s)
(16/22) type_specific.io-github-autotest-libvirt.virsh.desc.negative_test.invalid_domain_name: PASS (31.97 s)
(17/22) type_specific.io-github-autotest-libvirt.virsh.desc.negative_test.invalid_domain_uuid: PASS (37.48 s)
(18/22) type_specific.io-github-autotest-libvirt.virsh.desc.negative_test.invalid_option1: PASS (31.82 s)
(19/22) type_specific.io-github-autotest-libvirt.virsh.desc.negative_test.invalid_option2:^[[C PASS (52.56 s)
(20/22) type_specific.io-github-autotest-libvirt.virsh.desc.negative_test.invalid_option3: PASS (32.34 s)
(21/22) type_specific.io-github-autotest-libvirt.virsh.desc.negative_test.new_desc_without_string: PASS (31.11 s)
(22/22) type_specific.io-github-autotest-libvirt.virsh.desc.negative_test.transient_config: PASS (26.68 s)
TOTAL TIME: 855.56 s (14:15)
TESTS PASSED: 22
TESTS FAILED: 0
SUCCESS RATE: 100.00 %

File: libvirt/tests/src/virsh_cmd/domain/virsh_desc.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import logging
 import os
+import uuid
+import logging
 from autotest.client.shared import error
-from virttest import virsh
+from virttest import virsh, aexpect, libvirt_vm
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_desc.py]: You import libvirt_vm but never use it

File: libvirt/tests/src/virsh_cmd/domain/virsh_desc.py [hunk 2]
```diff
@@ -24,11 +32,36 @@ def run_cmd(name, options, desc_str, status_error):
...
-        status = cmd_result.exit_status
+        if "--edit" not in options:
+            cmd_result = virsh.desc(name, options, desc_str, ignore_status=True,
+                                    debug=True)
+            output = cmd_result.stdout.strip()
+            err = cmd_result.stderr.strip()
+            status = cmd_result.exit_status
+        else:
+            logging.debug("Setting domain desc \"%s\" by --edit"
+                          % desc_str)
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_desc.py]: s/ % /, /

**Comment by Chenditang** [comment]
@will-Do 
Updated, Thanks~~

**Comment by will-Do** [comment]
@Chenditang Well, you need to resolve the conflicts

**Comment by Chenditang** [comment]
@will-Do 
I think the conflict is resolved now, Thanks~~

File: libvirt/tests/src/virsh_cmd/domain/virsh_desc.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import logging
 import os
+import uuid
 
 from autotest.client.shared import error
 
-from virttest import virsh
+from virttest import virsh, aexpect
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_desc.py]: aexpect is not under virttest in avocado-vt now, need pip install aexpect and import aexpect here

**Comment by will-Do** [comment]
Looks good, merge this pr.

**Pull request merged**